### ISSUE TITLE
Paletted textures

### ISSUE THREAD

**Issue by ChillyWillyGuru** [Kazade/GLdc#19]
I'd like to request 4 and 8-bit paletted textures. That would make certain things much easier. Think of sprite based games - most use 16 color or 256 color sprites. Now you wouldn't have to convert the sprites to 15/16 bit mode to use them.

**Comment by Kazade** [comment]
Right, the way this is going to work is by implemented [EXT_paletted_texture](https://people.freedesktop.org/~marcheu/extensions/EXT/paletted_texture.html) and [EXT_shared_texture_palette](https://people.freedesktop.org/~marcheu/extensions/EXT/shared_texture_palette.html)

Need to do the following things:

 - Add a `TexturePalette` struct
 - Add `TexturePalette* palette` to `TextureObject`
 - Add `static TexturePalette* GLOBAL_PALETTE` to texture.c
 - Implement `glColorTableEXT`
 - Manipulate the PVR appropriately when textures are bound
 - Add support for `GL_COLOR_INDEX*_EXT` formats to `glTexImage2D`
 - Support `GL_TEXTURE_2D` and `SHARED_TEXTURE_PALETTE_EXT ` as targets to `glColorTableEXT`
 - Report both extensions as supported in GL_EXTENSIONS.
